Empire/Empire.cpp

```diff
@@ -107,6 +112,38 @@ namespace {
         return retval;
     }
 
+    float CalculateNewStockpile(int empire_id, int stockpile_location_id, float starting_stockpile,
+                                         const std::map<std::set<int>, float>& available_pp,
+                                         const std::map<std::set<int>, float>& allocated_pp,
+                                         const std::map<std::set<int>, float>& allocated_stockpile_pp)
+    {
+        const Empire* empire = GetEmpire(empire_id);
+        if (!empire) {
+            ErrorLogger() << "CalculateStockpileContribution() passed null empire.  doing nothing.";
+            return 0.0f;
+        }
+        float stockpile_transfer_efficiency = empire->GetMeter("METER_IMPERIAL_PP_TRANSFER_EFFICIENCY")->Current();
+        // locally, within same resource group as the stockpile location (empire capital) we currently allow stockpile contribution at no penalty
+        float local_stockpile_transfer_rate = 1.0;
```

Comment:
> As discussed, remove, or at least disable, the concept of a stockpile location

Hmm, I hadn't understood that brief exchange to go so far as you are saying now.  Stockpile location is in a fair bit of preexisting code; it seems to me that the discussions are still at a bit of an early stage to go through and rip that all out.

Also, I view my role here as more limited, at least currently-- @agrrr3 was working on the imperial stockpile and had on our forums requested help getting the completion projections to work properly; since that was one of my areas of past work, and since it seemed (to me at least) like the imperial stockpile was a good candidate for being a significant part of the Stealth revamp which I understood to be the focus for the next release, I thought I should take on the task of getting a suitable foundation for the imperial stockpile with working projections, which I think I've done. (I'll be happy to still follow up on the various bits of code grooming you've pointed to and any bugfixes.)  Overall, though, I am hoping that @agrrr3 will resume activity and run with the baton.

In the meantime, I'd like to suggest we have some more discussion on the forums about the stockpile now that there is a working base for people to play around with.